File: ai/agents/analyst/analyst_agent.py
import json
from json_repair import repair_json
from llm.ollama_client import OllamaClient
from .prompts import ANALYST_PROMPT

class AnalystAgent:

    @staticmethod
    def analyze(text: str):

        prompt = f"""
        {ANALYST_PROMPT}

        Regulation:
        {text}
        """

        response = OllamaClient.analyze(prompt)

        print("RAW LLM RESPONSE:")
        print(response)
        
        fixed = repair_json(response)
        
        print("REPAIRED JSON:")
        print(fixed)
        
        return json.loads(fixed)

File: backend/src/ai/analyst/analyst.agent.ts
import { ANALYST_PROMPT } from "./analyst.prompt.js";

import { OllamaClient } from "../llm/ollama.js";

export class AnalystAgent {
  static async analyze(text: string) {
    const prompt = `
${ANALYST_PROMPT}

Regulation:

${text}
`;

    const response = await OllamaClient.generate(prompt);

    return JSON.parse(response);
  }
}
